src/commands/data/import/tree.ts

```diff
@@ -45,23 +46,61 @@ export default class Import extends SfCommand<ImportResult[]> {
     }),
   };
 
-  public async run(): Promise<ImportResult[]> {
+  public async run(): Promise<ImportResult[] | undefined> {
     const { flags } = await this.parse(Import);
 
     const conn = flags['target-org'].getConnection(flags['api-version']);
-    const results = flags.plan
-      ? await importFromPlan(conn, flags.plan)
-      : await importFromFiles(conn, flags.files ?? []);
 
-    this.table({
-      data: results,
-      columns: [
-        { key: 'refId', name: 'Reference ID' },
-        { key: 'type', name: 'Type' },
-        { key: 'id', name: 'ID' },
-      ],
-      title: 'Import Results',
-    });
-    return results;
+    try {
+      const results = flags.plan
+        ? await importFromPlan(conn, flags.plan)
+        : await importFromFiles(conn, flags.files ?? []);
+
+      this.table({
+        data: results,
+        columns: [
+          { key: 'refId', name: 'Reference ID' },
+          { key: 'type', name: 'Type' },
+          { key: 'id', name: 'ID' },
+        ],
+        title: 'Import Results',
+      });
+      return results;
+    } catch (err) {
+      const error = err as SfError;
+      if (
+        error.cause &&
+        error.cause instanceof Error &&
+        'data' in error.cause &&
+        isObject(error.cause.data) &&
+        'message' in error.cause.data
```

Comment:
move all these checks to a small function like this:
https://github.com/salesforcecli/sf-plugins-core/blob/2bc43599313191af6df93cccf70fcfe0082c768f/src/errorHandling.ts#L60

see: https://www.typescriptlang.org/docs/handbook/2/narrowing.html